### PR TITLE
VisualStudio build fixes

### PR DESCRIPTION
**Pull request by tautschnig**
Fixes for #531 and #532. These are untested.

### CODE REVIEW

**Comment by forejtv** [comment]
Now it says:
```
remove_static_init_loops.cpp(72): error C2039: 'max': is not a member of 'std'
C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE\iomanip(18): note: see declaration of 'std'
remove_static_init_loops.cpp(72): error C3861: 'max': identifier not found
```
Googling tells me that `#include<algorithm>` might help.

(I am still trying to figure out how to show AppVeyor build results automatically)

**Comment by tautschnig** [comment]
Thanks a lot for the feedback, another commit to address std::max is now included. Could you please re-try?

File: src/goto-programs/goto_program_template.h [hunk 1]
```diff
@@ -216,8 +216,13 @@ class goto_program_templatet
     }
 
     //! Uniquely identify an invalid target or location
```
Comment (forejtv) [inline src/goto-programs/goto_program_template.h]: "...unless we are on old MS compiler"
(?)

**Comment by tautschnig** [comment]
The build appears to have succeeded! There is a ton of warnings that are eventually worth looking at (mostly about size_t <-> unsigned implicit conversions).

File: src/goto-programs/goto_program_template.h [hunk 2]
```diff
@@ -216,8 +216,13 @@ class goto_program_templatet
...
+    #if (defined _MSC_VER && _MSC_VER <= 1800)
```
Comment (forejtv) [inline src/goto-programs/goto_program_template.h]: Can you add an explanation for 1800 and the variable name? Something like "_MSC_VER is a version of the Microsoft 'cl' compiler, 1800 corresponds to the Visual Studio 2013 version"

Comment (tautschnig) [inline src/goto-programs/goto_program_template.h]: Let me know whether you're happy with one that I have just pushed!

**Pull request merged**